lms/djangoapps/grades/course_grade.py

```diff
@@ -294,14 +293,7 @@ def attempted(self):  # lint-amnesty, pylint: disable=invalid-overridden-method
         Returns whether any of the subsections in this course
         have been attempted by the student.
         """
-        if assume_zero_if_absent(self.course_data.course_key):
-            return True
-
-        for chapter in self.chapter_grades.values():
-            for subsection_grade in chapter['sections']:
-                if subsection_grade.all_total.first_attempted:
-                    return True
-        return False
+        return True
```

Comment:
I don't like this change at all. I understand logically, this is correct. However, this makes the meaning of the `attempted` property useless. Can we go to all places where this particular property is referenced and update the logic to not use the property? That way, we can remove this property all together.

Reply:
I thanked about it. But this function just override this [method from CourseGradeBase](https://github.com/openedx/edx-platform/pull/30978/files#diff-f0d17d91bec07c249a69bfeb1b4be40078e8c433136af33644f429eeddcab9b3R56). CourseGradeBase is patent for ZeroCourseGrade and CourseGrade, so this method return `False` for ZeroCourseGrade.